src/core/Provider.ts

```diff
@@ -41,7 +40,7 @@ const InnerProvider: React.FC<{
   if (isReactExperimental && r.current === defaultContextUpdate) {
     r.current = (f) => contextUpdate(f)
   }
-  return createElement(Fragment, {}, children)
+  return (children as ReactElement) ?? null
```

Comment:
```suggestion
  return (children ?? null) as ReactNode
```
How about this? We need to import `ReactNode`.

Reply:
That's not good as `ReactNode` is `ReactChild | ReactFragment | ReactPortal | boolean | null | undefined`, including `undefined` and `boolean` which are not valid return values.

Reply:
Oh, I see.

Reply:
Does that mean `<Provider>{true}</Provider>` is invalid?

Reply:
I'm sorry I was wrong, booleans are fine (although they render nothing), `undefined` is not.

Reply:
And even `undefined` is fine when being a child, like `<Provider>{undefined}</Provider>`. What is not fine is returning it from a component.

Reply:
Okay, thanks for the clarification. I guess InnerProvider is tricky.